lib/sprockets/utils/gzip.rb

```diff
@@ -0,0 +1,58 @@
+module Sprockets
+  module Utils
+    class Gzip
+      # Private: Generates a gzipped file based off of reference file.
+      def initialize(asset)
+        @content_type  = asset.content_type
+        @mtime         = asset.mtime
+        @source        = asset.source
+        @charset       = asset.charset
+      end
+
+      # Private: Returns whether or not an asset can be compressed.
+      #
+      # We want to compress any file that is text based.
+      # You do not want to compress binary
+      # files as they may already be compressed and running them
+      # through a compression algorithm would make them larger.
+      #
+      # Return Boolean.
+      def can_compress?(mime_types)
+        # The "charset" of a mime type is present if the value is
+        # encoded text. We can check this value to see if the asset
+        # can be compressed.
+        return true if @charset
+
+        # Svg images are text but do not have
+        # a charset defined, this is special cased.
+        return true if @content_type == "image/svg+xml".freeze
```

Comment:
This method could be writte as `@charset || @content_type == "image/svg+xml".freeze` plus the comments... only feedback, if you prefer this implementation fine.